### DISCUSSION TITLE
Vector256.Create messing with upper bits

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/runtime#53967] [General]
Not seeing why this happens in C# in both .Net 5/6, but not .Net Core 3.1:

```javascript
    var v = Vector256.Create(1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f);      //<1, 2, 3, 4, 5, 6, 7, 8>
    v = Vector256.Create(1.0f);                                                    //<1, 1, 1, 1, 0, 0, 0, 7.17E-43>
    v = Vector256.Create(-1.0f, -2.0f, -3.0f, -4.0f, -5.0f, -6.0f, -7.0f, -8.0f);  //<-1, -2, -3, -4, -5, -6, -7, -8>
    v = Vector256.Create(1.0f);                                                    //<1, 1, 1, 1, 1, 1, 1, 1>
```
The upper bits of line 2 should have matched line 4. However, moving the code around makes it occur with line 1.

**Comment by SingleAccretion** [comment]
What versions of .NET 5/6 does this reproduce on? Could you try the latest `main` from [`dotnet/installer`](https://github.com/dotnet/installer) and/or the latest .NET 5 version (`5.0.7`)? A bug similar to this was fixed at some point.

How does the exact reproduction look like (are you inspecting values under the debugger?)?

For reference, the following:
```cs
var v = Vector256.Create(1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f);    
Console.WriteLine(v);
v = Vector256.Create(1.0f);                                                  
Console.WriteLine(v);
v = Vector256.Create(-1.0f, -2.0f, -3.0f, -4.0f, -5.0f, -6.0f, -7.0f, -8.0f);
Console.WriteLine(v);
v = Vector256.Create(1.0f);                                                  
Console.WriteLine(v);
```
Prints the expected values for me on 44f050acc0f7791d6cd7ac772945444912bcf299.

**Reply by ghost** [reply]
Thank you, I am using the latest .Net 5/6. It does work correctly when not using the debugger or use .Net Core 3.1.

**Only when I set a breakpoint on any or all of the create lines then each one has bad output.**

I like to watch intrinsic methods carefully since they are tricky, but I can ignore this issue by just not setting the breakpoint on the Vector256 instruction.

**Reply by SingleAccretion** [reply]
So I take it this reproduces on the latest `main`, but only under the debugger? Interesting.

**Reply by tannergooding** [reply]
@davmason, @tommcdon.

**Reply by davmason** [reply]
@hoyosjs have you seen this?

**Reply by hoyosjs** [reply]
No, this would be new. And the debugger has issues handling Vector already... Creating an issue... Cc:@sdmaclea as this looks like instruction decoding, which we covered when he generated look up tables.

**Reply by hoyosjs** [reply]
Opened #54025
